core/packages/contracts/contracts/utils/Bitfield.sol

```diff
@@ -48,87 +41,103 @@ library Bitfield {
         bitfield = new uint256[](prior.length);
         uint256 found = 0;
 
-        for (uint256 i = 0; found < n; i++) {
+        for (uint256 i = 0; found < n;) {
             bytes32 randomness = keccak256(abi.encodePacked(seed, i));
-            uint256 index = uint256(randomness) % length;
+            uint256 index;
+            unchecked {
+                index = uint256(randomness) % length;
+            }
+
+            Location memory loc = toLocation(index);
 
             // require randomly selected bit to be set in prior
-            if (!isSet(prior, index)) {
+            if (!isSet(prior, loc)) {
+                unchecked { ++i; }
                 continue;
             }
 
             // require a not yet set (new) bit to be set
-            if (isSet(bitfield, index)) {
+            if (isSet(bitfield, loc)) {
+                unchecked { ++i; }
                 continue;
             }
 
-            set(bitfield, index);
+            set(bitfield, loc);
 
-            found++;
+            unchecked { ++found; }
+            unchecked { ++i; }
         }
 
         return bitfield;
     }
 
     function createBitfield(uint256[] calldata bitsToSet, uint256 length)
```

Comment:
This function does not need to be optimized at all since it is never called from on any on-chain code. So I wouldn't bother using `unchecked` inside it.

Reply:
Should we move these test helper functions to their own library that doesn't get deployed?

Reply:
Actually it is used in production by the relayer, which calls it via this function on BeefyClient: https://github.com/Snowfork/snowbridge/blob/da5bf982ec38d90e714e48cb9f357a782c40f59c/core/packages/contracts/contracts/BeefyClient.sol#L511

I'm open to removing `createBitfield` if we re-implement it in Golang and Typescript.

Reply:
Right, it's used before submitInitial. Not sure what the savings are, but I'm happy to move it out of Solidity 👍